spring-cloud-commons/src/main/java/org/springframework/cloud/client/ServiceInstance.java

```diff
@@ -71,4 +77,16 @@ default String getScheme() {
 		return null;
 	}
 
+	/**
+	 * @return The {@link String} value from instance metadata corresponding to the
+	 * {@link ServiceInstance#ZONE} key.
+	 */
+	default String getZone() {
```

Comment:
I'd rather not pollute `ServiceInstance` with shortcuts for metadata. I think this can go safely in the `ZonePreferenceServiceInstanceListSupplier`